app/src/main/java/com/fivegmag/a5gmsmediastreamhandler/MediaSessionHandlerAdapter.kt

```diff
@@ -223,6 +233,40 @@ class MediaSessionHandlerAdapter() {
         }
     }
 
+    fun reportConsumption() {
+        if (!bound) return
+
+        var location = TypedLocation("111","222")
+        val consumptionReportingUnits = ConsumptionReportingUnit(
+            "mediaConsumed",
+            EndpointAddress("ipv4Addr", "ipv6Addr", 80u),
+            "startTime",50, arrayOf(location)
+            )
+        var  consumptionData = ConsumptionReporting("mediaPlayerEntry","Consumption testing Data by Shilin", arrayOf(consumptionReportingUnits));
```

Comment:
We need to send the real data instead of some fake data:

* mediaPlayerEntry: Use the current MPD url
* reportingClientId: We can use the message identifier: `msg.sendingUid`
* mediaConsumed: Use the current Representation ID 
* startTime: Time when the switch occured
* duration: From my understanding: If multiple switches occurred use timestamp in between these switches. 

Starting point:
````
    override fun onDownstreamFormatChanged(
        eventTime: AnalyticsListener.EventTime,
        mediaLoadData: MediaLoadData
    ) {

        val to: String? = mediaLoadData.trackFormat?.id
       val startTime: Use the current timestamp here
    }
````